src/Microsoft.AspNet.Buffering/HttpBufferingFeature.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Buffering
+{
+    internal class HttpBufferingFeature : IHttpBufferingFeature
```

Comment:
You can check HttpResponse.HasStarted or HttpResponse.Body.CanSeek to know if the response body is buffered and/or if the headers can still be modified. Then Set Position and SetLength 0 zero to clear the body.

That said, MVC probably shouldn't render serialization exceptions inline. Throw it and let the exception handling logic decide.